.github/workflows/changelog-cd.yaml

```diff
@@ -0,0 +1,134 @@
+---
+name: Changelog generator CD
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - tools/changelog
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+**-changelog"
```

Comment:
when using multiple Go modules in a repository, the tag format should be <subdirs>/<version>: https://go.dev/wiki/Modules#faqs--multi-module-repositories

So in this case, this should be `"tool/changelog/v[0-9]+.[0-9]+.[0-9]+"`